File: 0121-Best-Time-to-Buy-and-Sell-Stock/0121.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = 0, profit = 0, tempProfit = 0;

        for (int i = 0; i < prices.length; i++) {
            if (i == 0) {
                buy = prices[i];
                continue;
            }

            if (prices[i] < buy)
                buy = prices[i];

            if (prices[i] > buy)
                tempProfit = prices[i] - buy;

            if (tempProfit > profit)
                profit = tempProfit;

            tempProfit = 0;
        }

        return profit;
    }
}

File: 0121-Best-Time-to-Buy-and-Sell-Stock/0121.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = 0
        temp_profit = 0
        profit = 0

        for i in range(len(prices)):
            if i == 0:
                buy = prices[i]
                continue
            
            if prices[i] < buy:
                buy = prices[i]

            if prices[i] > buy:
                temp_profit = prices[i] - buy

            if temp_profit > profit:
                profit = temp_profit
            
            temp_profit = 0
            
        return profit